samples/react-js-app/src/components/dashboard.tsx

```diff
@@ -35,11 +35,38 @@ export const Dashboard: FunctionComponent<null> = (): ReactElement => {
 
     const [ isAuth ] = useRecoilState(authState);
     const [ displayNameState ] = useRecoilState(displayName);
+    const [ isLoggedOut, setIsLoggedOut ] = useState(false);
+
+    useEffect(() => {
+        auth.on(Hooks.SignOut, () => {
+            setIsLoggedOut(true);
+        });
+    }, [auth]);
 
     return <div className="wrapper">
         <div className="menu">
-            <button onClick={ () => { history.push(SIGN_IN); } }>Sign In</button>
-            <button onClick={ () => { history.push(SIGN_OUT); } }>Sign Out</button>
+            <button onClick={ () => {
+                if (isAuth) {
+                    alert("You have already signed in!");
+                } else {
+                    setIsLoggedOut(false);
+                    history.push(SIGN_IN);
+                }
+            } }>
+                Sign In
+            </button>
+
+            <button onClick={ () => {
+                if (isAuth) {
+                    history.push(SIGN_OUT);
+                } else {
+                    alert("You haven't signed in!");
+                }
+            }
+            }>
+                Sign Out
```

Comment:
Let's add a localization placeholder here.

Reply:
This is a sample app. Do we need to support localization here?

Reply:
Yeah if this app is there just to try out the SDK I guess we don't need to support localization.